drivers/dma/dma_cavs_gpdma.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_cavs_gpdma
+
+#define GPDMA_CTL_OFFSET 0x004
+#define GPDMA_CTL_FDCGB BIT(0)
+
+#include "dma_dw_common.h"
+
+#define LOG_LEVEL CONFIG_DMA_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dma_cavs_gpdma);
+
+
+/* Device run time data */
+struct cavs_gpdma_data {
+	struct dw_dma_dev_data dw_data;
+};
+
+/* Device constant configuration parameters */
+struct cavs_gpdma_cfg {
+	struct dw_dma_dev_cfg dw_cfg;
+	uint32_t shim;
+};
+
+/* Disables automatic clock gating (force disable clock gate) */
+static void cavs_gpdma_clock_enable(const struct device *dev)
```

Comment:
Since it looks like the only delta vs. the main driver is this clock enable step, another design option here would be to use the existing driver unaltered and put the clock handling into a power management callback set up at the soc level.

Reply:
There's other alterations needed once I add linear link positioning and linked list transfers as well as some individual hardware soc workarounds I need to port over from sof. Some of that will be here, some of that will be in the common bits with some ifdefry (has to be I think)